### PR TITLE
CA-289189: Add sole PIF name changing action

### PR DESCRIPTION
**Pull request by jijiang**
Signed-off-by: Ji Jiang <ji.jiang@citrix.com>

### CODE REVIEW

File: XenModel/Actions/Network/SetSecondaryManagementPurposeAction.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+namespace XenAdmin.Actions
+{
+    public class SetSecondaryManagementPurposeAction : AsyncAction
+    {
+        private Pool pool;
+        private List<PIF> pifs;
+        private static readonly log4net.ILog log = log4net.LogManager.GetLogger(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType);
+
+        public SetSecondaryManagementPurposeAction(IXenConnection connection, Pool pool, List<PIF> pifs)
```
Comment (kc284) [inline XenModel/Actions/Network/SetSecondaryManagementPurposeAction.cs]: Any RBAC considerations for this action? Renaming does not sound particularly risky as an operation, but perhaps it should be subject to the same restrictions as the other re-configuration operations?

File: XenAdmin/Dialogs/NetworkingProperties.cs [hunk 2]
```diff
@@ -496,6 +497,13 @@ private void AcceptBtn_Click(object sender, EventArgs e)
             // list originally so that they can be used as a filter against downPIFs.
             newPIFs.RemoveAll(delegate(PIF p) { return !p.Changed; });
 
+            newNamePIFs.RemoveAll(delegate(PIF p) { return !p.Changed; });
```
Comment (kc284) [inline XenAdmin/Dialogs/NetworkingProperties.cs]: Let's modernise a bit this code and the 2 lines above it by using lambdas.

File: XenModel/Messages.resx [hunk 3]
```diff
@@ -771,6 +771,9 @@
   <data name="ACTION_SCANNING_SR_FROM" xml:space="preserve">
     <value>Performing SR scan from '{0}'</value>
   </data>
+  <data name="ACTION_SET_SECONDARY_MANAGEMENT_PURPOSE_TITLE" xml:space="preserve">
+    <value>Changing server networking interface name</value>
```
[the remark this answers is not in the document]
Comment (jijiang) [reply XenModel/Messages.resx]: Is "..." needed at the end? The string will appear on the Events tab -> Message column, not inside the status bar.

File: XenAdmin/Dialogs/NetworkingProperties.cs (hunk 2 above)
Comment (jijiang) [reply XenAdmin/Dialogs/NetworkingProperties.cs]: will fix it.

File: XenModel/Actions/Network/SetSecondaryManagementPurposeAction.cs (hunk 1 above)
Comment (jijiang) [reply XenModel/Actions/Network/SetSecondaryManagementPurposeAction.cs]: Actually in this action, only PIF.SaveChanges calling to server side is made. Is it enough to add this one to the RBAC precheck?

File: XenModel/Messages.resx (hunk 3 above)
[the remark this answers is not in the document]
Comment (kc284) [reply XenModel/Messages.resx]: I can't rememebr what otehr messages on there do, if the others don't have ellipsis, feel free to drop it.

File: XenModel/Actions/Network/SetSecondaryManagementPurposeAction.cs (hunk 1 above)
Comment (kc284) [reply XenModel/Actions/Network/SetSecondaryManagementPurposeAction.cs]: TYes, ok, the actual API call used is pif.set_other_config so let's add this one.

**Pull request merged**